test/image/mocks/axes_category_ascending.json

```diff
@@ -0,0 +1,13 @@
+{
+    "data": [{
+        "x": ["c","a","e","b","d"],
+        "y": [15,11,12,13,14]}
+    ],
+    "layout": {
+        "xaxis": {
+            "title": "category ascending",
+            "type": "category",
+            "categorymode": "category ascending",
+            "categorylist": ["y","b","x","a","d","z","e","c", "q", "k"]
```

Comment:
@etpinard the question is for @cpsievert but quick notes on why it fell this way:

1. The axis tick ordering and the line tracing are orthogonal concepts
2. The issue writeup from @cpsievert talked about missing points in some traces rather than unpredictably ordered points and when I raised the concern (i.e. trace order vs tick order) and the proposed solution here, there were no objections from anyone
3. The current solution allows snail trail diagrams, like the NYT chart http://www.nytimes.com/imagepages/2010/05/02/business/02metrics.html
4. This leaves the room for a separate feature: optionally specifying the trace order.